File: algorithms/cpp/646.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int findLongestChain(vector<vector<int>> &pairs) {
        int n = pairs.size();
        if (n == 0) return 0;
        sort(pairs.begin(), pairs.end());
        int result = 1, end = pairs[0][1];
        for (int i = 1; i < n; i++) {
            if (pairs[i][0] > end) {
                result += 1;
                end = pairs[i][1];
            }
            else
                end = min(end, pairs[i][1]);
        }
        return result;
    }
};

int main() {
    return 0;
}

File: algorithms/rust/646.rs
impl Solution {
    #[allow(dead_code)]
    pub fn find_longest_chain(mut pairs: Vec<Vec<i32>>) -> i32 {
        pairs.sort();
        let mut dp = vec![1; pairs.len()];

        let mut rslt = 1;
        for i in 0..pairs.len() {
            for j in i..pairs.len() {
                if pairs[j][0] > pairs[i][1] {
                    dp[j] = dp[j].max(dp[i] + 1);
                    rslt = rslt.max(dp[j]);
                }
            }
        }

        rslt
    }
}

struct Solution;

fn main() {
    println!("Hello, world!");
}
